src/components/common/button/Button.tsx

```diff
@@ -0,0 +1,37 @@
+import { observer } from 'mobx-react';
+import React from 'react';
+
+import './button.scss';
+
+interface ButtonProps {
+  onClick: () => void;
+  buttonText: string;
+  className?: string;
+  exiting?: boolean;
+  loading?: boolean;
+}
+
+@observer
+export class Button extends React.PureComponent<ButtonProps> {
+  public render() {
+    const { onClick, className, exiting, loading, buttonText } = this.props;
+
+    const btnText = loading ? 'loading' : buttonText;
+    const exitingClass = exiting ? 'exiting' : '';
+    const buttonClass = loading ? 'loading' : className;
```

Comment:
you'd want to apply the given class name at all times, regardless of loading - custom class names should control things like width/height, which shouldn't change with loading